### PR TITLE
Sirius Upgrade + Deprecation Replacement

### PR DESCRIPTION
**Pull request by sabieber**
Fixes: [SIRI-580](https://scireum.myjetbrains.com/youtrack/issue/SIRI-580)

### CODE REVIEW

File: src/main/java/sirius/biz/importer/format/ImportDictionary.java [hunk 1]
```diff
@@ -607,8 +608,7 @@ protected Optional<FieldDefinition> resolveComputedField(String name) {
      * @return the normalized version of the field
      */
     protected static String normalize(String field) {
-        return Strings.reduceCharacters(field.toLowerCase())
-                    .replaceAll("[^\\p{L}\\d_]", "");
+        return StringCleanup.reduceCharacters(field.toLowerCase()).replaceAll("[^\\p{L}\\d_]", "");
```
Comment (idlira) [inline src/main/java/sirius/biz/importer/format/ImportDictionary.java:611]: hier könnte man das field.toLowerCase durch StringCleanup::lowerCase ersetzen

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/importer/format/ImportDictionary.java (hunk 1 above)
Comment (cschierle) [inline src/main/java/sirius/biz/importer/format/ImportDictionary.java]: This should better match the intended cleanup call structure.
Please check the suggestion for validity in the IDE before committing.
```suggestion
        return Strings.cleanup(field, StringCleanup::lowercase, StringCleanup::reduceCharacters, text -> text.replaceAll("[^\\p{L}\\d_]", ""));
```

File: src/main/java/sirius/biz/storage/layer2/URLBuilder.java [hunk 2]
```diff
@@ -609,7 +610,7 @@ private String computeAccessToken(String authToken) {
 
     private void appendAddonText(StringBuilder result) {
         if (Strings.isFilled(addonText)) {
-            result.append(Strings.reduceCharacters(NON_URL_CHARACTERS.matcher(addonText).replaceAll("-")));
+            result.append(StringCleanup.reduceCharacters(NON_URL_CHARACTERS.matcher(addonText).replaceAll("-")));
```
Comment (cschierle) [inline src/main/java/sirius/biz/storage/layer2/URLBuilder.java:613]: This should better match the intended cleanup call structure.
Please check the suggestion for validity in the IDE before committing.
```suggestion
            result.append(Strings.cleanup(addonText, text -> NON_URL_CHARACTERS.matcher(addonText).replaceAll("-"), StringCleanup::reduceCharacters));
```

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/storage/layer2/URLBuilder.java (hunk 2 above)
Comment (sabieber) [reply src/main/java/sirius/biz/storage/layer2/URLBuilder.java:613]: This is 1 to 1 was the replaced deprecated method calls internally 😅 Why is it even deprecated when its semantics changed or am I missing something??

Comment (cschierle) [reply src/main/java/sirius/biz/storage/layer2/URLBuilder.java:613]: Note the class JavaDoc in [`StringCleanup`](https://github.com/scireum/sirius-kernel/blob/0a9e8b37f2a9ce68a3cbb8526d717994b3ed9b90/src/main/java/sirius/kernel/commons/StringCleanup.java#L23):
```
/**
 * Provides various methods to clean-up or reduce strings.
 * <p>
 * Please note, that most of the functions provided here are intended to be used via {@code Strings.cleanup(..)} and
 * therefore provide no internal <tt>null</tt> check.
...

Comment (sabieber) [reply src/main/java/sirius/biz/storage/layer2/URLBuilder.java:613]: `@deprecated Use {@link StringCleanup#reduceCharacters(String)} or {@code Strings.cleanup(input, Cleanup::reduceCharacters)} instead`

File: src/main/java/sirius/biz/importer/format/ImportDictionary.java [hunk 3]
```diff
@@ -607,8 +608,10 @@ protected Optional<FieldDefinition> resolveComputedField(String name) {
      * @return the normalized version of the field
      */
     protected static String normalize(String field) {
-        return Strings.reduceCharacters(field.toLowerCase())
-                    .replaceAll("[^\\p{L}\\d_]", "");
+        return Strings.cleanup(field,
+                               StringCleanup::reduceCharacters,
+                               StringCleanup::lowercase,
```
[the remark this answers is not in the document]
Comment (sabieber) [reply src/main/java/sirius/biz/importer/format/ImportDictionary.java]: Order of these is pretty much irrelevant, as `Ü -> Uuml -> uuml` and `Ü -> ü -> uuml`. But I accepted the suggestion anyway.

**Review by idlira — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**